### PR TITLE
Fixes multirev

### LINKED ISSUE #20934: 1 Click antag make revs issues
(WEB REPORT BY: chazach REMOTE: 206.221.180.138:7777)
# Revision
c11556c666b091fd3474a65e011c00bc404f38b5

# Description
can poll someone who is already a rev
also when it finally did pick a new person it didnt even make them a head rev

### PR DESCRIPTION
**Pull request by ShiftyRail**
Closes #20934 

The first part of the report isn't actually a bug, if there is a rev faction active, it will try to add the latejoiner role, which is in such a case a normal (red) rev.

Tested.

### CODE REVIEW

File: code/modules/admin/verbs/one_click_antag.dm [hunk 1]
```diff
@@ -87,6 +87,9 @@ client/proc/one_click_antag()
 			count--
 			var/mob/living/carbon/human/H = pick(candidates)
 			candidates.Remove(H)
+			var/initRoleID = initial(F.initroletype.id)
+			if (locate(initRoleID) in H.mind.antag_roles) // Ex: a head rev being made a revolutionary.
+				return FALSE
```
Comment (MadmanMartian) [inline code/modules/admin/verbs/one_click_antag.dm]: Shouldn't it just be continue rather than return false?

Comment (ShiftyRail) [reply code/modules/admin/verbs/one_click_antag.dm]: That's very true

File: code/modules/admin/verbs/one_click_antag.dm [hunk 2]
```diff
@@ -87,6 +87,9 @@ client/proc/one_click_antag()
 			count--
 			var/mob/living/carbon/human/H = pick(candidates)
 			candidates.Remove(H)
+			var/initRoleID = initial(F.initroletype.id) // A string.
```
Comment (MadmanMartian) [inline code/modules/admin/verbs/one_click_antag.dm]: Why not F.initial_role?

Comment (ShiftyRail) [reply code/modules/admin/verbs/one_click_antag.dm]: Because I forgot this var existed

**Comment by Intigracy** [comment]
There should still be a way to automatically make a headrev

**Pull request merged**